### PR TITLE
docs: add basic docs for KeyValueRepository

### PR DESCRIPTION
**Pull request by raymondfeng**
Add basic docs for `KeyValueRepository]` using https://github.com/strongloop/loopback4-example-shopping/pull/4 as an example.

## Checklist

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

### CODE REVIEW

File: docs/site/Repositories.md [hunk 1]
```diff
@@ -291,7 +291,103 @@ Please See [Testing Your Application](Testing-Your-Application.md) section in
...
+  constructor(@inject('datasources.redis') ds: RedisDataSource) {
+    super(ShoppingCart, ds);
+  }
+}
+```
+
+### Perform Key Value Operations
+
+The KeyValueRepository provides a set of key based operations, such as `set`,
+`get`, `delete`, `expire`, `ttl`, and `keys`. See [KeyValueRepository] for a
```
Comment (bajtos) [inline docs/site/Repositories.md]: > `See [KeyValueRepository] for a`

Missing link?

File: docs/site/Repositories.md [hunk 2]
```diff
@@ -291,7 +291,103 @@ Please See [Testing Your Application](Testing-Your-Application.md) section in
...
+
+  // Retrieve a cart by its key
+  const result = await repo.get(cart1.userId);
+  console.log(result);
+});
+
+testKV();
+```
+
+[keyvaluerepository]:
```
[the remark this answers is not in the document]
Comment (raymondfeng) [reply docs/site/Repositories.md]: Argh, `prettier` converts it to be lower case :-(.

**Comment by raymondfeng** [comment]
Comments addressed. PTAL.

File: docs/site/Repositories.md (hunk 1 above)
Comment (virkt25) [reply docs/site/Repositories.md]: Fix this before landing this PR.

File: docs/site/Repositories.md [hunk 3]
```diff
@@ -291,7 +291,103 @@ Please See [Testing Your Application](Testing-Your-Application.md) section in
...
-## Persisting Data without Juggler [Using MySQL database]
+## Access KeyValue Stores
+
+We can now access key-value stores such as [redis](Redis) using the
+[KeyValueRepository].
+
+### Define a KeyValue Datasource
+
+We first need to define a datasource to configure the key-value store. For
+better flexibility, we spilt the datasource definition into two files.
```
[the remark this answers is not in the document]
Comment (marioestradarosa) [reply docs/site/Repositories.md]: typo: spilt change to split  or splited  ?

File: docs/site/Repositories.md [hunk 4]
```diff
@@ -291,7 +291,108 @@ Please See [Testing Your Application](Testing-Your-Application.md) section in
...
+We can now access key-value stores such as [Redis](https://redis.io/) using the
+[KeyValueRepository]((https://github.com/strongloop/loopback-next/blob/master/packages/repository/src/repositories/kv.repository.ts).
+
+### Define a KeyValue Datasource
+
+We first need to define a datasource to configure the key-value store. For
+better flexibility, we spilt the datasource definition into two files. The json
+file captures the configuration properties and it can be possibly overridden by
+dependency injection.
+
```
Comment (marioestradarosa) [inline docs/site/Repositories.md]: sorry to chime in, but wouldn't be good to add the following NOTE?.  Especially for new LB4 adopters?

 **NOTE:** 
The prefix **redis** is related to the name of the data source, so you can use any name you want here. Usually you use `lb4 datasource name` to generate it, where `name`  is the prefix mentioned above.

Comment (raymondfeng) [reply docs/site/Repositories.md]: The effective `name` is from https://github.com/strongloop/loopback4-example-shopping/blob/master/src/datasources/redis.datasource.json#L2.

File: docs/site/Repositories.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (raymondfeng) [reply docs/site/Repositories.md]: Good catch. Will fix.

File: docs/site/Repositories.md (hunk 4 above)
Comment (marioestradarosa) [reply docs/site/Repositories.md]: 👍 I didn't notice the context of the example-shopping.

**Pull request merged**